### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by FelipeEnne**

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Start :clap: However, some requirements needs to be implemented in order for us to approve the milestone. check :arrow_down: 


#### Game flow requirements

* [x] A loop is needed in this game (recommended is `while` but anything else will do)
* [x] Inside loop, All actions should be repeated for each player.
* [x] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

- **[SUGGESTION]**
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```
#### UI interactions:

- In this step not everything must work, the point is to make sure that students thought about everything listed below. They might do it only with inline comments.
* [x] Basic game flow is prepared:
  * [x] `[OPTIONAL]` game ask players for their names
  * [x] `[REQUIRED]` game tells which player turn it is
  * [x] `[REQUIRED]` game asks player to select from available moves
  * [x] `[REQUIRED]` game informs player if selected move is invalid
  * [x] `[REQUIRED]` game displays board after player move
  * [x] `[REQUIRED]` game informs player if selected move is a winning move
  * [x] `[REQUIRED]` game informs player if selected move is a draw move
  * [x] `[REQUIRED]` game repeats all actions for next player's move

#### General requirements

- **[REQUIRED]** Configure your repository so it shows code linter checks.

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
_**Ahmed Mahfoudh**_

File: bin/main.rb [hunk 1]
```diff
@@ -2,4 +2,136 @@
...
+  if a.winner(player2)
+    game_on = false
+    break
+  end
+
+  if a.draw
+    game_on = false
+    break
+  end
+end
```
Comment (stratospherique) [inline bin/main.rb]: - [x] The loop should perform the action for each player **individually**. Use `@turn` variable to switch to current player at the beginning of the loop.

- :interrobang: Why using `break` if you're setting `game_on = false`

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Progress @Team :clap: However, some changes are still to be made in order to get approval for this milestone. Check :arrow_down: 

- **[REQUIRED]** Check the code comment below, it should help you refactor your code.

- **[SUGGESTION]** Keep in mind that this milestone focuses on the game flow and UI interaction. The class "TicTacToe" contains in addition code related to the game logic (i.e board & board manipulation, validation) even if it's not finalized. I recommend that you use that code in a game class (that you're gonna implement in the next milestone) and leave only UI related code(game start messages & loop actions) in the `main.rb` because it's cleaner and easier to debug and maintain.

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Good work on this project, but we need to fix the issues below before moving on ⬇️** 

1. After I start the game and I choose coordinates, nothing gets displayed on the board, a player mark should be displayed on the location if it's already taken by the player, please implement the following:
* [x] `[REQUIRED]` game displays board after player move.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: :clap: Congrats, milestone approved :heavy_check_mark: you may merge and pass to the next :+1: :+1: :+1: 

Keep up the good work :muscle: 
_**Ahmed Mahfoudh**_

**Pull request merged**